### ISSUE TITLE
Migration dependencies

### ISSUE THREAD

**Issue by HannesP** [sequelize/umzug#142]
Is it possible to have several migration timelines, where a migration in one timeline is dependent on the progress in another timeline? By timeline I mean a series of migrations that relate to the same tool (is there a better term for this?).

For example, I might have one migration timeline for managing (adding, removing, changing settings etc) Apache Kafka topics, and another migration timeline for my app's database schema. I could also have a timeline for data that is dynamic but an integral part of the application.

For example, let's say that I have a timeline for my database. In migration 155, I add a data structure to store user roles, something along the lines of:

```
# Migration 155, schema timeline
# UP:
CREATE TABLE user_roles (name string, domain_id number, slug string, ...);
# DOWN:
DROP TABLE user_roles;
```

In my data timeline, I want to fill this with a number of roles, that _could_ change later, but that are to be considered as fairly constant:

```
# Migration 17, data timeline
# UP:
POST /user_roles/ {"name": "Admin", "domain_id": 1, "slug": "admin", ...}
POST /user_roles/ {"name": "User", "domain_id": 1, "slug": "user", ...}
POST /user_roles/ {"name": "Editor", "domain_id": 2, "slug": "editor", ...}
# DOWN:
DELETE /user_roles/admin
DELETE /user_roles/user
DELETE /user_roles/editor
```

From the above description, it is obvious that migration 17 in the data timeline cannot be run before migration 155 in the schema timeline has been run. Is there a way to deal with this kind of dependencies in umzug?

**Comment by tamlyn** [comment]
I have a need for a similar feature. It could potentially be implemented with a custom sorting function (#153).

**Comment by chadxz** [comment]
why not have two sets of migrations, with their state managed separately?
